File: docs/code-examples/disconnected_space_v2.cpp
// Disconnect two spaces.

#include <rerun.hpp>

namespace rr = rerun;

int main() {
    auto rr_stream = rr::RecordingStream("rerun-example-disconnected_space");
    rr_stream.connect("127.0.0.1:9876").throw_on_failure();

    // These two points can be projected into the same space..
    rr_stream.log("world/room1/point", rr::Points3D(rr::datatypes::Vec3D{0.0f, 0.0f, 0.0f}));
    rr_stream.log("world/room2/point", rr::Points3D(rr::datatypes::Vec3D{1.0f, 1.0f, 1.0f}));

    // ..but this one lives in a completely separate space!
    rr_stream.log("world/wormhole", rr::DisconnectedSpace(true));
    rr_stream.log("world/wormhole/point", rr::Points3D(rr::datatypes::Vec3D{2.0f, 2.0f, 2.0f}));
}

File: docs/code-examples/disconnected_space_v2.rs
//! Disconnect two spaces.

use rerun::{
    archetypes::{DisconnectedSpace, Points3D},
    MsgSender, RecordingStreamBuilder,
};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) =
        RecordingStreamBuilder::new("rerun_example_disconnected_space").memory()?;

    // These two points can be projected into the same space..
    MsgSender::from_archetype("world/room1/point", &Points3D::new([(0.0, 0.0, 0.0)]))?
        .send(&rec)?;
    MsgSender::from_archetype("world/room2/point", &Points3D::new([(1.0, 1.0, 1.0)]))?
        .send(&rec)?;

    // ..but this one lives in a completely separate space!
    MsgSender::from_archetype("world/wormhole", &DisconnectedSpace::new(true))?.send(&rec)?;
    MsgSender::from_archetype("world/wormhole/point", &Points3D::new([(2.0, 2.0, 2.0)]))?
        .send(&rec)?;

    rerun::native_viewer::show(storage.take())?;

    Ok(())
}
